Library/Homebrew/dev-cmd/irb.rb

```diff
@@ -21,15 +24,20 @@ module Homebrew
   module_function
 
   def irb
-    if ARGV.include? "--examples"
+    args = Homebrew::CLI::Parser.new do
```

Comment:
I was just thinking while reading this: what do you think about a refactoring that looks like this:
```ruby
    args = Homebrew::CLI::Parser.parse do
      switch "--examples"
      switch "--pry"
    end
```
or similar as you never seem to make use of the `Parser` object directly.

Reply:
Yeah, seems better & more readable this way. Will open a seperate PR for this.

Reply:
@GauthamGoli Cool. Feel free to combine both ideas above into same PR (although different commits ideally).

Reply:
@MikeMcQuaid I've added option for specifying environment variables and while adding a test for this, I remembered that I do make use of `Parser` Object directly in tests. It was written in existing way to make testing easier.

Reply:
@GauthamGoli Should still be able to have a `.parse` class method which calls `new` and then runs `parse` on it so it's a short-hand of the version here without needing to change the existing `new` or `#parse` methods. As a general principle I'd say a public API should never be less readable/harder to use just to make testing easier.

Reply:
Agree. Will do it. 👍